### PR TITLE
[SPARK-34172][SQL] Add `SHOW DATABASES` as table-valued function

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Add  show tables as Table-valued function make user convenient to analysis db information in SQL

### Why are the changes needed?
make user convenient to analysis db information in SQL

### How was this patch tested?
Added UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
ok to test

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala [hunk 1]
```diff
@@ -103,6 +106,46 @@ object ResolveTableValuedFunctions extends Rule[LogicalPlan] {
           "numPartitions" -> IntegerType) {
         case Seq(start: Long, end: Long, step: Long, numPartitions: Int) =>
           Range(start, end, step, Some(numPartitions))
+      }),
+
+    "show_namespaces" -> Map(
```
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala:111]: Please also update https://github.com/apache/spark/blob/6786500d49efc00a658a917cb6fbf9d87090a64b/docs/sql-ref-syntax-qry-select-tvf.md#L3

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala [hunk 2]
```diff
@@ -103,6 +106,46 @@ object ResolveTableValuedFunctions extends Rule[LogicalPlan] {
...
+      /* show_namespaces() */
```
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala:112]: This implementation will add a lot of TVFs. How about just add one TVF, something like this:
```sql
SELECT * FROM SELECT_FROM_COMMAND('show_namespaces') WHERE identifier='a' AND pattern = 'b';
```

Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala:112]: > This implementation will add a lot of TVFs. How about just add one TVF, something like this:
> 
> ```sql
> SELECT * FROM SELECT_FROM_COMMAND('show_namespaces') WHERE identifier='a' AND pattern = 'b';
> ```

In this way its hard to fit where condition to command. and always fetch all data then filter?

If it's show tables, we can't directly filter data in metadata level.

**Review by dongjoon-hyun — COMMENTED** [review]
Do other DBMSs support this as a built-in show_columns function? If then, could you update the PR description by adding those references?

**Comment by AngersZhuuuu** [comment]
> Do other DBMSs support this as a built-in show_columns function? If then, could you update the PR description by adding those references?

after a long time search, no other DBMS support this kind of functions. But it's really helpful for data analysis user to have a better insight  for metadata.

**Pull request closed without merging**